contributors/design-proposals/add-on/coredump-daemonset.md

```diff
@@ -0,0 +1,115 @@
+# coredump detector add-on
+
+Cao Shufeng (@CaoShuFeng)
+
+## Abstract
+
+This proposal aims at managing coredump files generated in cluster containers.
+
+## Motivation and Goals
+
+Application coredump is one of required feature for enterprise systems. In Kubernetes, we need to take "multi-tenancy" and "security" into account. Technically, under Linux, each server’s coredump can be controlled via [/proc/sys/kernel/core_pattern](http://man7.org/linux/man-pages/man5/core.5.html) of the host. By this, an application coredump can be passed into host’s special file or special program. We need to build API to specify
+- Where coredumps should be stored and access control to them
+  - typically, namespaced persistent storage rather than host’s file system
+  - list/get/delete coredump metadata
+  - access control to dumped data
+  - quota for coredump for each namespace
+  - no core dump label for specific environment (such as development environ)
+
+
+## Community Work
+
+- make core_pattern support namespace in kernel (not approved yet): https://lkml.org/lkml/2017/2/7/931
+
+## Proposed Design
+
+To implement this, we need to make the following components run in kubernetes cluster:
+1. A coredump-detector daemonset runs in each node. It will:
+   * put a new special program integrated with kuberentes’s API into /proc/sys/kernel/core_pattern
+   * determine process in which pod/namespace generated the core file
+   * save coredump metadata into api-server
+   * save coredump file into persistent volume
+2. A coredump-controller which support quota of coredump files in each namesapce
+![architecture](./architecture.png)
+
+# core_pattern
+To determine whether a core file is generated for process in a k8s container, we
+override /proc/sys/kernel/core_pattern kernel parameter in kubelet node.
+```
+|/coredump/coredump-detector -P=%P -p=%p -e=%e -t=%t -c=/coredump/config --log_dir=/coredump/
```

Comment:
Which mount namespace would this `/coredump` path belongs to? In particular, are you assuming that this is going to be in the pod-app mount namespace, in the kubelet mount namespace or in the kernel-init/host one?

Reply:
kernel does not support namespace for coredump.
So this would be kernel-init/host namespace.